src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -144,7 +144,8 @@ export class ConnectionDialogService implements IConnectionDialogService {
 					defaultProvider = filteredKeys[0];
 				}
 				else {
-					defaultProvider = keys[0];
+					// CMS isn't default unless called by the extension specifically
+					defaultProvider = keys.filter(provider => provider === Constants.cmsProviderName)[0];
```

Comment:
Is there a way to solve this without hard-coding CMS?  Perhaps we should do something like (1) try to look-up MSSQL directly, if that fails just return the first item.  

The general rule is (1) don't special-case provides and (2) if you must special-case a provider try to limit to MSSQL since that is the default/built-in provider.

Reply:
also, please update the comment block at the top of this method if the logic ends up being a little different than what's captured there.

Reply:
The issue is we can't take the first item as the providers as populated in random orders (based on extension loading). I'll try to look up MSSQL directly in this case, then.

Reply:
Also this doesn't seem to affect the above case where _params.providers has items. If that contained CMS we'd still try to default to that - is that what you want? I don't know offhand what this logic is doing exactly so I can't say for sure whether that's expected

---
In reply to: [290520917](https://github.com/microsoft/azuredatastudio/pull/5884#discussion_r290520917) [](ancestors = 290520917)

Reply:
@abist yeah, the suggestion is to (1) find MSSQL directly and then (2) if that fails take the first one.

Reply:
@abist in case it's not clear, looking up MSSQL may fail if that extension is uninstalled or disabled, for example, if we want to build a PGSQL-only bundle in the future.